tests/first_stage_test.py

```diff
@@ -7,6 +11,278 @@
 import testlib
 
 
+def own_create_child(args, blocking, pipe_size=None, preexec_fn=None):
+    """
+    Create a child process whose stdin/stdout/stderr is connected to a pipe.
+
+    :param list args:
+        Program argument vector.
+    :param bool blocking:
+        If :data:`True`, the pipes use blocking IO, otherwise non-blocking.
+    :param int pipe_size:
+        If not :data:`None`, use the values as the pipe size.
+    :param function preexec_fn:
+        If not :data:`None`, a function to run within the post-fork child
+        before executing the target program.
+    :returns:
+        :class:`PopenProcess` instance.
+    """
+    parent_rfp, child_wfp = mitogen.core.pipe(blocking=blocking)
+    child_rfp, parent_wfp = mitogen.core.pipe(blocking=blocking)
+    stderr_r, stderr = mitogen.core.pipe(blocking=blocking)
```

Comment:
Note to self: Confirmed passing non-blocking fds to `subprocess.Popen()` does result in non-blocking stdio fds in the child https://gist.github.com/moreati/edb3e4d631678a20c46419c1473ca625